core/embed/firmware/main.c

```diff
@@ -64,6 +66,7 @@ int main(void) {
 #endif
 
   collect_hw_entropy();
+  drbg_mix_hw_entropy();
```

Comment:
> Given that it's constant, I am not sure that it really contributes much to the DRBG.

 The purpose of mixing the hardware entropy into the drbg is to differentiate the state of the drbg across devices. So the hardware entropy plays role of a personalisation string. 

> Using the same secret for storage encryption and DRBG initialization is basically a form of key reuse.

But is that really a problem?  Can we allocate another block in the OTP memory for this purpose?

Reply:
I think I don't understand your arguments. On one hand you are saying that mixing potentially secret low-entropy coordinates of clicks into the drbg should be OK. On the other hand you are saying that mixing 256-bit high-entropy "secret", that is used also in the storage, into 256-bit drbg is not.

Reply:
> On one hand you are saying that mixing potentially secret low-entropy coordinates of clicks into the drbg should be OK.

I proposed we could mix the 2 or 3 least significant bits of each coordinate, which are highest in entropy and at the same time they say very little, if anything, about the actual position of the touch on the screen as a whole. (The ideal number of LSBs to use would require some analysis and I am questioning whether it's worth the work and implementation BTW.)

> On the other hand you are saying that mixing 256-bit high-entropy "secret", that is used also in the storage, into 256-bit drbg is not.

Precisely, because it's a form of key reuse, not to mention it contributes no randomness but is rather a personalization string.

To put it simply:
- The touch LSBs carry no secret information and are a source of randomness.
- The OTP secret contains secret information and is not a source of randomness.